lib/msal-common/src/cache/entities/AccountEntity.ts

```diff
@@ -150,7 +150,7 @@ export class AccountEntity {
                 ? idToken.claims.oid
                 : idToken.claims.sid;
             account.localAccountId = localAccountId;
-            account.username = idToken.claims.preferred_username;
+            account.username = idToken.claims.preferred_username || (idToken.claims.emails? idToken.claims.emails[0]: "");
```

Comment:
Any chance this is too opinionated (taking the first email)?

Reply:
We talked to the B2C team and they said in 99% of use cases there will only be 1 email ever returned. The reason it's an array is for the other 1% where someone has deliberately configured their custom policy to return more than 1. It's also one of the reasons we're specifically calling out that `getAccountById` should be considered more reliable than `getAccountByUsername`

Reply:
Makes sense, can you please add that as a comment here?

Reply:
Done